File: 0026-remove-duplicates-from-sorted-array/0026-remove-duplicates-from-sorted-array.kt
class Solution {
    fun removeDuplicates(nums: IntArray): Int {
        var previous = Int.MAX_VALUE
        var newEnd = 0
        for (i in nums.indices){
            if (nums[i] == previous) {
                nums[i] = 0
            } else {
                nums[newEnd] = nums[i]
                newEnd += 1
                previous = nums[i]
            }
        }    
        return newEnd 
    }
}

File: 0026-remove-duplicates-from-sorted-array/0026-remove-duplicates-from-sorted-array.rs
impl Solution {
    pub fn remove_duplicates(nums: &mut Vec<i32>) -> i32 {
        let mut temp: Vec<i32> = Vec::new();
        temp.push(nums[0]);
        for i in 1..nums.len() {
            if nums[i] != nums[i-1] {
                temp.push(nums[i]);
            }
        }
        for i in 0..temp.len() {
            nums[i] = temp[i];
        }
        temp.len() as i32
    }
}